lib/dao.js

```diff
@@ -2356,6 +2559,200 @@ DataAccessObject.prototype.unsetAttribute = function unsetAttribute(name, nullif
 };
 
 /**
+ * Replace set of attributes.
+ * Performs validation before replacing.
+ *
+ * @trigger `validation`, `save` and `update` hooks
+ * @param {Object} data Data to replace
+ * @param {Object} [options] Options for replaceAttributes
+ * @param {Function} cb Callback function called with (err, instance)
+ */
+
+DataAccessObject.prototype.replaceAttributes = function replaceAttributes(data, options, cb) {
+  var connectionPromise = stillConnecting(this.getDataSource(), this, arguments);
+  if (connectionPromise) {
+    return connectionPromise;
+  }
+
+  if (options === undefined && cb === undefined) {
+    if (typeof data === 'function') {
+      cb = data;
+      data = undefined;
+    }
+  } else if (cb === undefined) {
+    if (typeof options === 'function') {
+      cb = options;
+      options = {};
+    }
+  }
+
+  cb = cb || utils.createPromiseCallback();
+  options = options || {};
+
+  assert((typeof data === 'object') && (data !== null),
+    'The data argument must be an object');
+  assert(typeof options === 'object', 'The options argument must be an object');
+  assert(typeof cb === 'function', 'The cb argument must be a function');
+
+  var inst = this;
+  var Model = this.constructor;
+  var connector = inst.getConnector();
+  assert(typeof connector.replaceAttributes === 'function',
+    'replaceAttributes() must be implemented by the connector');
+
+  if (isPKMissing(Model, cb))
+    return cb.promise;
+
+  var allowExtendedOperators = connector.settings
+    && connector.settings.allowExtendedOperators;
+
+  var strict = this.__strict;
+  var model = Model.modelName;
+  var hookState = {};
+
+  // Convert the data to be plain object so that update won't be confused
+  if (data instanceof Model) {
+    data = data.toObject(false);
+  }
+  data = removeUndefined(data);
+
+  // Make sure id(s) cannot be changed
+  var idNames = Model.definition.idNames();
+  for (var i = 0, n = idNames.length; i < n; i++) {
+    var idName = idNames[i];
+    if (data[idName] !== undefined && !idEquals(data[idName], inst[idName])) {
+      var err = new Error('id property (' + idName + ') ' +
+        'cannot be updated from ' + inst[idName] + ' to ' + data[idName]);
+      err.statusCode = 400;
+      process.nextTick(function() {
+        cb(err);
+      });
+      return cb.promise;
+    }
+  }
+
+  var context = {
+    Model: Model,
+    where: byIdQuery(Model, getIdValue(Model, inst)).where,
+    data: data,
```

Comment:
Again, because this operation modified the instance as whole, you should pass `ctx.instance` instead of `ctx.data` + `ctx.where` to "before save" hook.